### PR TITLE
feat: add kn-py-slack example

### PR DESCRIPTION
**Pull request by sgrman**
## Summary

Please describe your changes and what you plan to introduce as part of this pull request submission. If it fixes a bug or resolves a feature request, be sure to link to the issue numbers following the template below.

## Pull Request Checklist
🚨 Please review the [guidelines for contributing](https://vmweventbroker.io/community) to this repository.

- [x] Please ensure that you are making a pull request against the **Development** branch
- [ ] Please use the `WIP` keyword in the title of your PR if you are not ready for review
- [ ] Please ensure that you have opened a Github Issue if you are resolving/fixing a problem
- [x] Please ensure that you have [signed](https://help.github.com/en/github/authenticating-to-github/signing-commits) all commits and that you have [squashed](https://medium.com/@slamflipstrom/a-beginners-guide-to-squashing-commits-with-git-rebase-8185cf6e62ec) all relevant commits related to your change
- [x] Please make sure that you have tested your change locally by successfully [building and deploying the VMware Event Broker Appliance](https://vmweventbroker.io/kb/contribute-appliance) and/or [building and deploying VMware Event Router](https://vmweventbroker.io/kb/contribute-eventrouter)
- [x] Please include any relevant screenshots and/or output as part of your testing
- [x] Please include any documentation updates that is applicable for your changes

## Change Type

What types of changes does your code introduce to the VMware Event Broker Appliance?

_Put an `x` in all boxes that apply_

Please check the type of change your PR introduces:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation changes
- [ ] Other (please describe):

## Resolved Issues

List of Issues closed or resolved by this PR. Add multiple `Closes` keyword followed by the issue number (e.g. Closes #ISSUE-NUMBER)

Closes #ISSUE-NUMBER

## Testing Verification

* Re-did code example for kn-py-slack.  Aligned documentation to conform to other examples.

* Please include any relevant screenshots and/or output as part of the testing
Screenshots of running local tests are in the README as examples to follow.

## Additional Information

I had to abandon my previous PR #419  because I forgot to rename my own branch and it created some divergance that I could not undo.  This is from my fork, branch sgr-development, to the main development branch.

If you have any questions/comments, feel free to reach out to team on Slack [#vcenter-event-broker-appliance](https://vmwarecode.slack.com/archives/CQLT9B5AA)

Thank you from the VEBA Team! 🥳

### CODE REVIEW

File: examples/knative/python/kn-py-slack/README.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+# kn-ps-slack
```
Comment (embano1) [inline examples/knative/python/kn-py-slack/README.md:1]:
```suggestion
# kn-py-slack
```

File: examples/knative/python/kn-py-slack/README.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+ * Environment: development
+ * Debug mode: on
+ * Running on all addresses.
+   WARNING: This is a development server. Do not use it in a production deployment.
+ * Running on http://172.17.0.2:8080/ (Press CTRL+C to quit)
+ * Restarting with stat
+ * Debugger is active!
+ ```
+ 
+In a separate terminal window, go to the test directory and use the ```testevent.json``` file to validate the function is working. You should see output similar to this below.
```
Comment (embano1) [inline examples/knative/python/kn-py-slack/README.md:40]:
```suggestion
In a separate terminal window, go to the test directory and use the `testevent.json` file to validate the function is working. You should see output similar to this below.
```

File: examples/knative/python/kn-py-slack/README.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+```
+curl -i -d@testevent.json localhost:8080
+
+HTTP/1.1 100 Continue
+
+HTTP/1.0 204 NO CONTENT
```
Comment (embano1) [inline examples/knative/python/kn-py-slack/README.md:47]: Needs update with handler changes

**Review by embano1 — CHANGES_REQUESTED** [review]

File: examples/knative/python/kn-py-slack/README.md (hunk 1 above)
Comment (sgrman) [reply examples/knative/python/kn-py-slack/README.md]: made the change and signed commit 9d9afc8.

File: examples/knative/python/kn-py-slack/README.md (hunk 2 above)
Comment (sgrman) [reply examples/knative/python/kn-py-slack/README.md]: made the change and signed commit 9d9afc8.

File: examples/knative/python/kn-py-slack/README.md (hunk 3 above)
Comment (sgrman) [reply examples/knative/python/kn-py-slack/README.md]: made the change and signed commit 9d9afc8.

File: examples/knative/python/kn-py-slack/README.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+```console
+cd test
+```
+
+Update the `docker-test-env-variable` file with your Slack webook URL.
+
+Start the container image by running the following command:
+
+```console
+docker run -e FUNCTION_DEBUG=true -e PORT=8080 --env-file docker-test-env-variable -it --rm -p 8080:8080 <docker-username>/kn-py-slack:1.0
```
Comment (embano1) [inline examples/knative/python/kn-py-slack/README.md:28]:
```suggestion
docker run -e PORT=8080 --env-file docker-test-env-variable -it --rm -p 8080:8080 <docker-username>/kn-py-slack:1.0
```

File: examples/knative/python/kn-py-slack/README.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+Content-Length: 3
+Server: Werkzeug/1.0.1 Python/3.9.0
+Date: Tue, 25 May 2021 13:59:31 GMT
+```
+Return to the previous terminal window where you started the docker image, and you should see something similar to the following:
+
+```
+2021-05-25 08:59:28,400 INFO werkzeug MainThread :  * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
+2021-05-25 08:59:31,423 DEBUG urllib3.connectionpool Thread-1 : Starting new HTTPS connection (1): hooks.slack.com:443
+2021-05-25 08:59:31,614 DEBUG urllib3.connectionpool Thread-1 : https://hooks.slack.com:443 "POST /services/T024JFTN4/B01U92186ES/rAAyiTbJFc2gNLe91X7ZjumO HTTP/1.1" 200 22
```
Comment (embano1) [inline examples/knative/python/kn-py-slack/README.md:58]: Is this your token there?

File: examples/knative/python/kn-py-slack/README.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+```console
+docker push <docker-username>/kn-py-slack:1.0
+```
+
+Update the `slack_secret.json` file with your Slack webhook configurations and then create the kubernetes secret which can then be accessed from within the function by using the environment variable named called `SLACK_SECRET`.
+
+```console
+# create secret
+
```
Comment (embano1) [inline examples/knative/python/kn-py-slack/README.md:83]: Remove whitespace

File: examples/knative/python/kn-py-slack/README.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+```
+
+Edit the `function.yaml` file with the name of the container image from Step 1 if you made any changes. If not, the default VMware container image will suffice. By default, the function deployment will filter on the `VmPoweredOffEvent` vCenter Server Event. If you wish to change this, update the `subject` field within `function.yaml` to the desired event type.
+
+
+Deploy the function to the VMware Event Broker Appliance (VEBA).
+
+```console
+# deploy function
+
```
Comment (embano1) [inline examples/knative/python/kn-py-slack/README.md:97]: Whitespace

File: examples/knative/python/kn-py-slack/README.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+annotations:
+  autoscaling.knative.dev/maxScale: "1"
+  autoscaling.knative.dev/minScale: "1"
+```
+
+# Step 4 - Undeploy
+
+```console
+# undeploy function
+
```
Comment (embano1) [inline examples/knative/python/kn-py-slack/README.md:113]: Whitespace

**Review by embano1 — CHANGES_REQUESTED** [review]

**Review by embano1 — APPROVED** [review]
THX SO MUCH! LGTM

**Pull request merged**